### PR TITLE
improvement: Add the basic framework for an ete test and add a test for checking static compaction 

### PR DESCRIPTION
**Pull request by k-simons**
- Add the basic framework for an ete test and add a test for checking static compaction 
- Pull out common test configuration, helper functions, and test set up into a common test file

## After this PR

==COMMIT_MSG==
==COMMIT_MSG==

### CODE REVIEW

File: cmd/integration/server_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+						"memory": ToResource(resource.MustParse("4096Mi")),
+					},
+				},
+			},
+		},
+		Status: v1beta2.ResourceReservationStatus{
+			Pods: map[string]string{
+				"driver":     "my-pod-driver",
+				"executor-1": "my-pod-executor-1",
+				"executor-2": "missing-pod-1",
```
Comment (Chrisbattarbee) [inline cmd/integration/server_test.go:75]: Are we aiming to test the case of a dead executors reservation being re-used by a new executor?

If we're just looking to test the happy case, we can remove this.

File: cmd/integration/server_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				Namespace: "podNamespace",
+				Labels: map[string]string{
+					common.SparkRoleLabel:  common.Executor,
+					common.SparkAppIDLabel: "appID1",
+				},
+				Annotations: map[string]string{
+					common.ExecutorCPU:    "2",
+					common.ExecutorMemory: "4096Mi",
+					common.ExecutorCount:  "4",
+				},
```
Comment (Chrisbattarbee) [inline cmd/integration/server_test.go:176]: Executors dont have these annotations normally so we should probably remove these here

File: cmd/integration/common.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/palantir/k8s-spark-scheduler/cmd"
+	config2 "github.com/palantir/k8s-spark-scheduler/config"
```
[the remark this answers is not in the document]
Comment (k-simons) [reply cmd/integration/common.go:23]: Done

File: cmd/integration/common.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+			},
+		}).
+		WithDisableGoRuntimeMetrics().
+		WithInitFunc(func(ctx context.Context, initInfo witchcraft.InitInfo) (func(), error) {
+			rootCtx = ctx
+			f := func(ctx context.Context) error {
+				var err error
+				ref, err = cmd.InitServerWithClients(ctx, initInfo, allClients)
+				require.NoError(t, err)
+				return err
```
[the remark this answers is not in the document]
Comment (k-simons) [reply cmd/integration/common.go:64]: Done

File: cmd/integration/common.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+			}
+			err := wapp.RunWithFatalLogging(ctx, f)
+			require.NoError(t, err)
+			return nil, err
```
[the remark this answers is not in the document]
Comment (k-simons) [reply cmd/integration/common.go:68]: Done

File: cmd/integration/common.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+		for _, crd := range crds.Items {
+			crd.Status.Conditions = []apiextensionsv1.CustomResourceDefinitionCondition{
+				{
+					Type:   apiextensionsv1.Established,
+					Status: apiextensionsv1.ConditionTrue,
+				},
+			}
+			_, err = allClients.APIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Update(context.Background(), &crd, metav1.UpdateOptions{})
+			require.NoError(t, err)
+		}
```
[the remark this answers is not in the document]
Comment (k-simons) [reply cmd/integration/common.go:88]: Yeah I tweaked this to do it separately, its easier to understand now. Done

File: cmd/integration/common.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	return TestSetup{
+		ref:     ref,
+		ctx:     rootCtx,
+		cleanup: cleanup,
+	}
+}
+
+// ToResource de-references a Quantity
```
[the remark this answers is not in the document]
Comment (k-simons) [reply cmd/integration/common.go:105]: Done

File: go.mod [hunk 8]
```diff
@@ -26,6 +26,8 @@ require (
 	sigs.k8s.io/controller-runtime v0.11.0
 )
 
+require github.com/stretchr/testify v1.8.0
```
[the remark this answers is not in the document]
Comment (k-simons) [reply go.mod:29]: Done

File: cmd/integration/common.go [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+func ToResource(parse resource.Quantity) *resource.Quantity {
+	return &parse
+}
+
+func waitForCondition(ctx context.Context, t *testing.T, condition func() bool) {
```
[the remark this answers is not in the document]
Comment (k-simons) [reply cmd/integration/common.go:110]: Re-use it where?

File: cmd/integration/server_test.go (hunk 2 above)
Comment (k-simons) [reply cmd/integration/server_test.go:176]: Done

File: cmd/integration/server_test.go (hunk 1 above)
Comment (k-simons) [reply cmd/integration/server_test.go:75]: No, this is testing the case where a pod used to exists, and then got re-scheduled

**Review by Chrisbattarbee — APPROVED** [review]

**Pull request merged**